api/src/main/java/io/opentelemetry/context/propagation/HttpExtractor.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.context.propagation;
+
+import io.opentelemetry.context.Context;
+import javax.annotation.Nullable;
+
+public interface HttpExtractor {
+  <C> Context extract(Context ctx, C carrier, Getter<C> getter);
+
+  interface Getter<C> {
+    @Nullable
```

Comment:
As discussed in our meeting today, This interface should have another method that provides access to all the headers:
```java
Iterable<String> headers();
```